ui-core/src/components/inputs/datePicker/CalendarPicker.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import cx from 'classnames';
+import { ChevronLeft, ChevronRight } from '@osrd-project/ui-icons';
+import Calendar from './Calendar';
+import { CalendarSlot } from './type';
+import useCalendarPicker from './useCalendarPicker';
+
+export type CalendarPickerProps = {
+  initialDate?: Date;
+  selectedSlot?: CalendarSlot;
+  selectableSlot?: CalendarSlot;
+  numberOfMonths?: 1 | 2 | 3;
+  mode?: 'single' | 'range';
+  onDateChange?: (slot: CalendarSlot | undefined) => void;
+  modalPosition: {
+    top: number;
+    left: number;
+  };
+  calendarPickerRef: React.RefObject<HTMLDivElement>;
+};
+
+const CalendarPicker: React.FC<CalendarPickerProps> = ({
+  initialDate,
+  selectedSlot: initialSelectedSlot,
+  selectableSlot,
+  numberOfMonths = 1,
+  mode = 'single',
+  onDateChange = () => {},
```

Comment:
Don't set a default value for onDateChange
```suggestion
  onDateChange,
```